web_src/js/components/RepoContributors.vue

```diff
@@ -0,0 +1,360 @@
+      dateFrom: null,
+      dateUntil: null,
+      startDate: null,
+      endDate: null,
+    };
+  },
+  mounted() {
+    this.fetchGraphData();
+
+    $('.ui.dropdown').dropdown({
+      onChange: (val) => {
+        this.dateFrom = this.startDate;
+        this.dateUntil = this.endDate;
+        this.type = val;
+        this.sortContributors();
+      }
+    });
+  },
+  methods: {
+    sortContributors() {
+      const contributors = this.filterContributorWeeksByDateRange();
+      const sortingCriteria = `total_${this.type}`;
+      this.sortedContributors = Object.values(contributors).filter((contributor) => contributor[sortingCriteria] !== 0).sort((a, b) =>
+        a[sortingCriteria] > b[sortingCriteria] ? -1 : a[sortingCriteria] === b[sortingCriteria] ? 0 : 1
+      ).slice(0, 100);
+    },
+    async fetchGraphData() {
+      this.isLoading = true;
+      try {
+        const response = await fetch(`${this.repoLink}/activity/contributors/data`);
+        const data = await response.json();
+        const {total, ...rest} = data;
+        this.contributorsStats = rest;
+        this.dateFrom = new Date(total.weeks[0].week);
+        this.dateUntil = new Date();
+        this.startDate = this.dateFrom;
+        this.endDate = this.dateUntil;
+        this.sortContributors();
+        this.totalStats = total;
+        this.errorText = '';
+      } catch (e) {
+        this.errorText = e.message;
+      } finally {
+        this.isLoading = false;
+      }
+    },
+
+    filterContributorWeeksByDateRange() {
+      const filteredData = {};
+
+      const data = this.contributorsStats;
+      for (const key of Object.keys(data)) {
+        const user = data[key];
+        user['total_commits'] = 0;
+        user['total_additions'] = 0;
+        user['total_deletions'] = 0;
+        user['max_contribution_type'] = 0;
+        const filteredWeeks = user.weeks.filter((week) => {
+          const weekDate = new Date(week.week);
+          if (weekDate >= this.dateFrom && weekDate <= this.dateUntil) {
+            user['total_commits'] += week.commits;
+            user['total_additions'] += week.additions;
+            user['total_deletions'] += week.deletions;
+            if (week[this.type] > user['max_contribution_type']) {
+              user['max_contribution_type'] = week[this.type];
+            }
+            return true;
+          }
+          return false;
+        });
+
+        filteredData[key] = {...user, weeks: filteredWeeks};
+      }
+
+      return filteredData;
+    },
+
+    maxMainGraph() {
+      const maxValue = Math.max(
+        ...this.totalStats.weeks.map((o) => o[this.type])
+      );
+      const [cooefficient, exp] = maxValue
+        .toExponential()
+        .split('e')
+        .map(Number);
+      if (cooefficient % 1 === 0) {
+        return maxValue;
+      }
+      return (1 - (cooefficient % 1)) * 10 ** exp + maxValue;
+    },
+    maxContributorGraph() {
+      const maxValue = Math.max(
```

Comment:
I explained it above. If someone knows a way to make Y value fixed I can replace it with these.

Reply:
> Then why isn't this calculation simply `return ...this.sortedContributors.map((c) => c['max_contribution_type'])`?

Because it would mean returning 15.423 for maximum Y value for the graphs which looks really bad. This method is responsible for rounding up that value to 20.000 which looks nice :)

I also explained this in the comments.